tools/c7n_azure/c7n_azure/provider.py

```diff
@@ -41,9 +56,39 @@ def initialize_policies(self, policy_collection, options):
         return policy_collection
 
     def get_session_factory(self, options):
+        base_url = self.get_cloud_base_url()
+
         return partial(Session,
                        subscription_id=options.account_id,
-                       authorization_file=options.authorization_file)
+                       authorization_file=options.authorization_file,
+                       base_url=base_url)
+
+    def _set_cloud(self, options):
+        if self.cloud:
+            return
+
+        cloud_list = options.get('regions')
+
+        if not cloud_list:
+            self.cloud = AZURE_PUBLIC_CLOUD
+            return
+
+        # Only support passing in one cloud at a time
+        cloud = self.region_to_cloud.get(cloud_list[0])
```

Comment:
We can log an error. I think it may be easy to be confused because with AWS you can pass in multiple regions at once and we are overloading the term to be used per Cloud